pipeline/src/lib.rs

```diff
@@ -46,21 +48,23 @@ pub fn access_element<T: FieldElement>(
 }
 
 #[allow(clippy::print_stdout)]
-pub fn inputs_to_query_callback<T: FieldElement>(inputs: Vec<T>) -> impl QueryCallback<T> {
+pub fn inputs_to_query_callback<T: FieldElement>(
+    inputs: HashMap<u32, Vec<T>>,
+) -> impl QueryCallback<T> {
     move |query: &str| -> Result<Option<T>, String> {
         // TODO In the future, when match statements need to be exhaustive,
         // This function probably gets an Option as argument and it should
         // answer None by Ok(None).
 
         match &parse_query(query)?[..] {
-            ["\"input\"", index] => access_element("prover inputs", &inputs, index),
+            ["\"input\"", index] => access_element("prover inputs", &inputs[&0], index),
             ["\"data\"", index, what] => {
                 let what = what
-                    .parse::<usize>()
+                    .parse::<u32>()
                     .map_err(|e| format!("Error parsing what: {e})"))?;
                 assert_eq!(what, 0);
```

Comment:
should this be removed now?

Reply:
This function (`inputs_to_query_callback`) is used to create callbacks to input vectors (like we did before). We currently have this hacky convention that the input vector is passed as channel 0 in the HashMap version